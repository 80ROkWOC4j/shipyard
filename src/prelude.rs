pub use crate::get::GetComponent;
pub use crate::iter::{iterators, IntoIter, Shiperator};
pub use crate::not::Not;
pub use crate::remove::Remove;
pub use crate::run::System;
#[doc(hidden)]
pub use crate::run::SystemData;
pub use crate::sparse_set::{sort, sort::Sortable, View, ViewMut};
pub use crate::storage::{AllStorages, Entities, EntitiesMut, Key};
pub use crate::world::World;
pub use crate::Unique;
#[doc(hidden)]
#[cfg(feature = "proc")]
pub use shipyard_proc::system;

#[cfg(feature = "parallel")]
pub use crate::ThreadPool;
