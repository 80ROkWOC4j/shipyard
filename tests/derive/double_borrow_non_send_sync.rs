use shipyard::prelude::*;

struct NonSendSyncStruct(core::marker::PhantomData<*const ()>);

#[system(NonSendSyncSys)]
fn run(_: NonSendSync<&NonSendSyncStruct>, _: NonSendSync<&mut NonSendSyncStruct>) {}

#[system(NonSendSyncSysUnique)]
fn run(_: NonSendSync<&NonSendSyncStruct>, _: Unique<NonSendSync<&mut NonSendSyncStruct>>) {}

#[system(UniqueNonSendSyncSys)]
fn run(_: Unique<NonSendSync<&NonSendSyncStruct>>, _: NonSendSync<&mut NonSendSyncStruct>) {}

#[system(UniqueUnique)]
fn run(_: Unique<NonSendSync<&NonSendSyncStruct>>, _: Unique<NonSendSync<&mut NonSendSyncStruct>>) {}

fn main() {}
