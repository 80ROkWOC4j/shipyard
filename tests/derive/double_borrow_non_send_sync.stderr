error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
 --> $DIR/double_borrow_non_send_sync.rs:6:8
  |
6 | fn run(_: NonSendSync<&NonSendSyncStruct>, _: NonSendSync<&mut NonSendSyncStruct>) {}
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
 --> $DIR/double_borrow_non_send_sync.rs:9:8
  |
9 | fn run(_: NonSendSync<&NonSendSyncStruct>, _: Unique<NonSendSync<&mut NonSendSyncStruct>>) {}
  |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/double_borrow_non_send_sync.rs:12:8
   |
12 | fn run(_: Unique<NonSendSync<&NonSendSyncStruct>>, _: NonSendSync<&mut NonSendSyncStruct>) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Cannot borrow again a storage already borrowed exclusively, you may want to remove the shared borrow.
  --> $DIR/double_borrow_non_send_sync.rs:15:8
   |
15 | fn run(_: Unique<NonSendSync<&NonSendSyncStruct>>, _: Unique<NonSendSync<&mut NonSendSyncStruct>>) {}
   |        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
